### ISSUE TITLE
":TSUpdate" raises errors after 9a7301de4c452889fb59369b75cc445eeeae60d8

### ISSUE THREAD

**Issue by spindensity** [nvim-treesitter/nvim-treesitter#2186]
### Before reporting

- [ ] I have updated my neovim version to latest _master_
- [X] I have updated my plugin to the latest version
- [X] I have run `:TSUpdate`
- [X] I have read the [troubleshooting section](https://github.com/nvim-treesitter/nvim-treesitter#troubleshooting)

### Describe the bug

* Neovim: 0.6.1 stable;
* OS: Windows 11.
* nvim-treesitter: master branch 479fde3471ad6833d950abf645c0ebece15eaea4

Execute `:TSUpdate`, the following errors occur:

![image](https://user-images.githubusercontent.com/33576822/148019657-6daa9f13-abe8-46e9-9200-653bea17601c.png)

### To Reproduce

1. Open neovim;
2. Execute `:TSUpdate`.

### Expected behavior

No errors occur.

### Output of `:checkhealth nvim-treesitter`

```markdown
nvim-treesitter: require("nvim-treesitter.health").check()
========================================================================
## Installation
  - OK: `tree-sitter` found  0.20.2 (parser generator, only needed for :TSInstallFromGrammar)
  - OK: `node` found v16.13.1 (only needed for :TSInstallFromGrammar)
  - OK: `git` executable found.
  - OK: `clang` executable found. Selected from { "clang", "cl" }
    Version: clang version 13.0.0
  - OK: Neovim was compiled with tree-sitter runtime ABI version 13 (required >=13). Parsers must be compatible with runtime ABI.

## Parser/Features H L F I J
  - regex          ✓ . . . . 
  - yang           ✓ . ✓ . . 
  - make           ✓ . . . ✓ 
  - jsonc          ✓ ✓ ✓ ✓ ✓ 
  - scss           ✓ . . ✓ . 
  - pug            ✓ . . . ✓ 
  - dockerfile     ✓ . . . ✓ 
  - scala          ✓ . ✓ . ✓ 
  - go             ✓ ✓ ✓ ✓ ✓ 
  - toml           ✓ ✓ ✓ ✓ ✓ 
  - typescript     ✓ ✓ ✓ ✓ ✓ 
  - dart           ✓ ✓ . ✓ ✓ 
  - tlaplus        ✓ . ✓ . ✓ 
  - hcl            ✓ . ✓ ✓ ✓ 
  - supercollider  ✓ ✓ ✓ ✓ ✓ 
  - prisma         ✓ . . . . 
  - http           ✓ . . . ✓ 
  - c_sharp        ✓ ✓ ✓ . ✓ 
  - llvm           ✓ . . . . 
  - ocamllex       ✓ . . . ✓ 
  - hjson          ✓ ✓ ✓ ✓ ✓ 
  - python         ✓ ✓ ✓ ✓ ✓ 
  - ocaml_interface✓ ✓ ✓ . ✓ 
  - pioasm         ✓ . . . ✓ 
  - ocaml          ✓ ✓ ✓ . ✓ 
  - cpp            ✓ ✓ ✓ ✓ ✓ 
  - surface        ✓ . ✓ ✓ ✓ 
  - elixir         ✓ ✓ ✓ ✓ ✓ 
  - erlang         . . . . . 
  - json5          ✓ . . . ✓ 
  - bibtex         ✓ . ✓ ✓ . 
  - vim            ✓ ✓ . . ✓ 
  - bash           ✓ ✓ ✓ . ✓ 
  - zig            ✓ . ✓ ✓ ✓ 
  - css            ✓ . ✓ ✓ ✓ 
  - html           ✓ ✓ ✓ ✓ ✓ 
  - kotlin         ✓ . . . ✓ 
  - javascript     ✓ ✓ ✓ ✓ ✓ 
  - cmake          ✓ . ✓ . . 
  - php            ✓ ✓ ✓ ✓ ✓ 
  - r              ✓ ✓ . . . 
  - rst            ✓ ✓ . . ✓ 
  - ruby           ✓ ✓ ✓ ✓ ✓ 
  - ledger         ✓ . ✓ ✓ ✓ 
  - devicetree     ✓ ✓ ✓ ✓ ✓ 
  - latex          ✓ . ✓ . ✓ 
  - perl           ✓ . . . . 
  - turtle         ✓ ✓ ✓ ✓ ✓ 
  - nix            ✓ ✓ ✓ . ✓ 
  - beancount      ✓ . ✓ . . 
  - dot            ✓ . . . ✓ 
  - tsx            ✓ ✓ ✓ ✓ ✓ 
  - gowork         ✓ . . . ✓ 
  - glsl           ✓ ✓ ✓ ✓ ✓ 
  - svelte         ✓ . ✓ ✓ ✓ 
  - gomod          ✓ . . . ✓ 
  - godot_resource ✓ ✓ ✓ . . 
  - cuda           ✓ ✓ ✓ ✓ ✓ 
  - heex           ✓ . ✓ ✓ ✓ 
  - sparql         ✓ ✓ ✓ ✓ ✓ 
  - gdscript       ✓ ✓ . . ✓ 
  - commonlisp     ✓ ✓ ✓ . . 
  - query          ✓ ✓ ✓ ✓ ✓ 
  - clojure        ✓ ✓ ✓ . ✓ 
  - jsdoc          ✓ . . . . 
  - json           ✓ ✓ ✓ ✓ . 
  - comment        ✓ . . . . 
  - yaml           ✓ ✓ ✓ ✓ ✓ 
  - glimmer        ✓ . . . . 
  - pascal         ✓ ✓ ✓ ✓ ✓ 
  - lua            ✓ ✓ ✓ ✓ ✓ 
  - rust           ✓ ✓ ✓ ✓ ✓ 
  - verilog        ✓ ✓ ✓ . ✓ 
  - java           ✓ ✓ . ✓ ✓ 
  - c              ✓ ✓ ✓ ✓ ✓ 
  - ql             ✓ ✓ . ✓ ✓ 
  - fish           ✓ ✓ ✓ ✓ ✓ 
  - graphql        ✓ . . ✓ ✓ 
  - teal           ✓ ✓ ✓ ✓ ✓ 
  - vue            ✓ . ✓ ✓ ✓ 
  - julia          ✓ ✓ ✓ ✓ ✓ 
  - fennel         ✓ ✓ . . ✓ 

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang}
```

### Output of `nvim --version`

```text
NVIM v0.6.1
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compiled by runneradmin@fv-az152-430

Features: -acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM\sysinit.vim"
  fall-back for $VIM: "C:/Program Files/nvim/share/nvim"

Run :checkhealth for more info
```

**Comment by theHamsta** [comment]
I've seen this before that GitHub doesn't serve certain revisions. Can you try `require'nvim-treesitter.install'.prefer_git = false`?

**Comment by spindensity** [comment]
> I've seen this before that GitHub doesn't serve certain revisions. Can you try `require'nvim-treesitter.install'.prefer_git = false`?

I think the problem is caused by https://github.com/nvim-treesitter/nvim-treesitter/commit/9a7301de4c452889fb59369b75cc445eeeae60d8:

https://github.com/nvim-treesitter/nvim-treesitter/blob/9a7301de4c452889fb59369b75cc445eeeae60d8/lockfile.json#L71-L73

The commit `7eff728289fa991234a9300d46d7ce1fa5ab1b12` does not exist in the repo `https://gitlab.com/jirgn/tree-sitter-fusion.git` at all.

![image](https://user-images.githubusercontent.com/33576822/148086636-916f7326-6acc-415e-92e9-fc1fcad08281.png)

**Comment by theHamsta** [comment]
Sure it exists: https://gitlab.com/jirgn/tree-sitter-fusion/-/archive/7eff728289fa991234a9300d46d7ce1fa5ab1b12/tree-sitter-fusion-7eff728289fa991234a9300d46d7ce1fa5ab1b12.tar.gz . Otherwise, it wouldn't have passed the pipeline. It's however interesting that it only works when you trim the `.git` Maybe Gitlab changed it's API?

Git will only see revisions that are direct ancestors of branches that HEAD has.

**Comment by chenxiaolong** [comment]
I did a `--mirror` clone of https://gitlab.com/jirgn/tree-sitter-fusion and it looks like commit 7eff728289fa991234a9300d46d7ce1fa5ab1b12 refers to the `refs/merge-requests/17/head` ref. Is that expected? (It's weird because the Gitlab repo seems to have merge requests disabled.)

**Comment by theHamsta** [comment]
@jirgn any ideas why this happened?

**Comment by jirgn** [comment]
Hey,
I gave it a look and think know what happened. But do not know why it is a Problem on the gitlab API.
The  [MR](https://gitlab.com/jirgn/tree-sitter-fusion/-/merge_requests/17) squashed  commits and therefore the addressed hash is not present in the main branch. 
When I created the PR in this repo, i manually ran `update-lockfile.sh`. It looks like I did this once before final release of [tree-sitter-fusion](https://gitlab.com/jirgn/tree-sitter-fusion). 
This should be fixed with an lockfile update for fusion.

I did not notice it was a problem, because like @theHamsta noted it works on linux.
@theHamsta Is it bad practice to update the lockfile on a pull-request?

**Comment by theHamsta** [comment]
Usually, it is only necessary to update the lockfile when doing a breaking change. It can often reuse the commit generated by the update-lockfile-pr. This PR is currently blocked by some failing parser but it would usually have been solved by itself since we simply updated to the newester master/main.

I'm in favor for using curl also on Windows. Git is not guaranteed to find all commits and curl is simply faster since it does not have to check out the whole repository. There were problems for old versions on Windows but maybe we could just advice in the Windows wiki that people having problems with curl can just try out git.
